File: internal/cloudsync/markdown.go
package cloudsync

import "strings"

// NormalizeMarkdown converts CRLF and bare CR line endings to LF, the canonical
// form used at the local repository boundary before hashing or serializing.
func NormalizeMarkdown(s string) string {
	s = strings.ReplaceAll(s, "\r\n", "\n")
	return strings.ReplaceAll(s, "\r", "\n")
}

File: frontend/src/sync/markdown.js
// Markdown line-ending normalization for the browser sync port (R6.1). Matches
// internal/cloudsync/markdown.go: CRLF and bare CR become LF, the canonical form
// used at the wire boundary before hashing or serializing.
export function normalizeMarkdown(s) {
  s = s.replace(/\r\n/g, '\n')
  return s.replace(/\r/g, '\n')
}
